src/main/java/org/example/usersService/UserManagement.java

```diff
@@ -0,0 +1,67 @@
+package org.example.usersService;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class UserManagement {
+    private Map<String, User> users;
+    private Map<String, String> userInfoLogin;
+
+    public UserManagement(Map<String, User> users) {
+        this.users = users;
+    }
+
+    public void addUser(User user) {
+        users.put(user.getId(), user);
+        System.out.println("User added.");
+    }
+
+    public void deleteUser(User user) {
+        if (users.remove(user.getId()) == null)
+            throw new RuntimeException("User not found.");
+        System.out.println("User deleted.");
+    }
+
+    public void updateUser(User user) {
+        if (users.containsKey(user.getId()))
+        throw new RuntimeException("User not found.");
+        System.out.println("User updated");
+    }
+    public void displayUsers(User user) {
+        System.out.println("User List:");
+        users.values().forEach(user1 -> System.out.println("Username: " + user.getUsername() + "Password: " + user.getPassword()));
+    }
+
+    public void registerUser(String userLogin, String userPassword){
+        if (!userInfoLogin.containsKey(userLogin) || userInfoLogin .equals(null)){
```

Comment:
Here you should first check if userInfoLogin is null and if `!userInfoLogin.containsKey(userLogin)`. Currently in your code you first run the `containsKey` function, where if `userInfoLogin` is null you will get an exception.

Reply:
This should not be "||", it should be "&&". The user exists if the list is not null and it contains the key. Your logic is saying that the user exists if it has the key and is equal to null

Reply:
you can just do `userInfoLogin == null`. Also if userInfoLogin is null and you called .equals you will get a null pointer exception since you are trying to call equals method a null object.